src/Vector.cpp

```diff
@@ -0,0 +1,207 @@
+#include "Vector.h"
+
+
+// Constructor
+Vector::Vector(size_t size, int value)
+    : m_size(size)
+    , m_capacity(get_capacity(size))
+    , m_data(new int[m_capacity])
+{
+    // Initialize all elements to the given value
+    for (size_t i = 0; i < m_size; ++i) {
+        m_data[i] = value;
+    }
+}
+
+// Copy constructor
+Vector::Vector(const Vector& other)
+    : m_size(other.m_size)
+    , m_capacity(other.m_capacity)
+    , m_data(new int[m_capacity])
+{
+    // Copy all elements from the other vector
+    for (size_t i = 0; i < m_size; ++i) {
+        m_data[i] = other.m_data[i];
+    }
+}
+
+// Move constructor
+Vector::Vector(Vector&& other) noexcept
+    : m_size(other.m_size)
+    , m_capacity(other.m_capacity)
+    , m_data(other.m_data)
+{
+    // Take ownership of the other vector's data
+    other.m_size = 0;
+    other.m_capacity = 0;
+    other.m_data = nullptr;
+}
+
+// Copy assignment operator
+Vector& Vector::operator=(const Vector& other) {
+    if (this != &other) {
+        // Copy and swap idiom for exception safety
+        Vector temp(other);
+        std::swap(m_size, temp.m_size);
+        std::swap(m_capacity, temp.m_capacity);
+        std::swap(m_data, temp.m_data);
+    }
+    return *this;
+}
+
+// Move assignment operator
+Vector& Vector::operator=(Vector&& other) noexcept {
+    if (this != &other) {
+        // Free the old data and take ownership of the new data
+        delete[] m_data;
+        m_size = other.m_size;
+        m_capacity = other.m_capacity;
+        m_data = other.m_data;
+        other.m_size = 0;
+        other.m_capacity = 0;
+        other.m_data = nullptr;
+    }
+    return *this;
+}
+
+// append an element to the end of the vector
+void Vector::push_back(int value) {
+    if (m_size == m_capacity) {
+        if (m_capacity < 128) {
```

Comment:
This logic is duplicated from the `get_capacity` static method. Code duplication is not encouraged since it makes your code harder to maintain (i.e - you decide to change the capacity growth logic, so now you have to find all the logic implementation and change them).
Try to think about how you can reduce this duplication, maybe with small changes to `get_capacity` method